runtime/lua/vim/lsp.lua

```diff
@@ -844,6 +844,11 @@ function lsp.buf_notify(bufnr, method, params)
   end)
 end
 
+local line = nil;
+local pos = nil;
+local line_to_cursor = nil;
+local textMatch = nil;
```

Comment:
would be nice to put this in a "state bag" variable as a hint for readers.

```
local _state = {lines=nil, pos=nil, ...}
```

Then when these variables are used in the function it is a useful hint that this is some shared/long-lived state.

```
character = _state.pos[2]
```

It's also useful for debugging, one can inspect `_state` instead of juggling a bunch of variables.